src/world/Objects/Units/Players/Player.cpp

```diff
@@ -1244,6 +1244,18 @@ void Player::applyLevelInfo(uint32_t newLevel)
             BaseStats[i] = lvlinfo->Stat[i];
             CalcStat(i);
         }
+
+#if VERSION_STRING >= WotLK
+        // Recalculate Heilrooms
+        for (uint8_t i = 0; i < CURRENCYTOKEN_SLOT_END; ++i)
+        {
+            if (Item* pItem = getItemInterface()->GetInventoryItem(i))
+            {
+                if (i < INVENTORY_SLOT_BAG_END)      // only equipment slots get mods.
```

Comment:
Since you only apply item mods if this is true, why don't you just do "for (uint8_t i = 0; i < INVENTORY_SLOT_BAG_END; ++i)"?